src/Evolution/Systems/GeneralizedHarmonic/Equations.cpp

```diff
@@ -0,0 +1,255 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/GeneralizedHarmonic/Equations.hpp"
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/System.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+
+namespace GeneralizedHarmonic {
+
+template <size_t Dim>
+void ComputeDuDt<Dim>::apply(
+    const gsl::not_null<tnsr::aa<DataVector, Dim>*> dt_psi,
+    const gsl::not_null<tnsr::aa<DataVector, Dim>*> dt_pi,
+    const gsl::not_null<tnsr::iaa<DataVector, Dim>*> dt_phi,
+    const tnsr::aa<DataVector, Dim>& psi, const tnsr::aa<DataVector, Dim>& pi,
+    const tnsr::iaa<DataVector, Dim>& phi,
+    const tnsr::iaa<DataVector, Dim>& d_psi,
+    const tnsr::iaa<DataVector, Dim>& d_pi,
+    const tnsr::ijaa<DataVector, Dim>& d_phi, const Scalar<DataVector>& gamma0,
```

Comment:
isn't `d_phi` `iiaa` since the partial derivatives are symmetric as well? This does raise the question, can our differentiation infrastructure handle that?

Reply:
no, there is a constraint (the four-index constraint) that can be violated by d_i phi_jab != d_j phi_iab

Reply:
*shakes fist angrily at sky* We need that infinite precision floating point :stuck_out_tongue: